### PR TITLE
(maint) Minor PS tweaks in prep for pipe rewrite

### PR DESCRIPTION
**Pull request by Iristyle**
- Pulls extraneous output from Invoke-PowerShellUserCode
- Refactors event signaling

### CODE REVIEW

File: lib/puppet_x/templates/init_ps.ps1.erb [hunk 1]
```diff
@@ -501,19 +498,30 @@ function Invoke-PowerShellUserCode
...
+{
+  [CmdletBinding()]
+  param(
+    [String]
+    $EventName
+  )
+
+  $event = [System.Threading.EventWaitHandle]::OpenExisting($EventName)
+
+  [System.Diagnostics.Debug]::WriteLine("Signaling event $EventName")
```
Comment (Iristyle) [inline lib/puppet_x/templates/init_ps.ps1.erb]: Thoughts on whether to keep / remove the debug output?  This allows you to use SysInternals DbgView to monitor progress...

Comment (glennsarti) [inline lib/puppet_x/templates/init_ps.ps1.erb]: Sounds like the same problem with "Waited 50ms" debug log spam.  Perhaps jsut comment it out for the moment?

Comment (jpogran) [inline lib/puppet_x/templates/init_ps.ps1.erb]: Less so as you need to actively ask/listen for for those messages, its not output by default

Comment (Iristyle) [inline lib/puppet_x/templates/init_ps.ps1.erb]: It won't show up in any logs / for Puppet users.

If you just turn on DbgView, you'll see output from other apps flowing by (though that's usually turned on with a switch or config option)

Comment (glennsarti) [inline lib/puppet_x/templates/init_ps.ps1.erb]: Sorry...meant, same problem as; "How much debug output should we emit"?

Comment (glennsarti) [inline lib/puppet_x/templates/init_ps.ps1.erb]: I'm not really fussed either way.  If we keep it in, I would expect that we would properly instrument this with debug statements later.

**Comment by jpogran** [comment]
👍

**Comment by Iristyle** [comment]
FYI - I just added one more refactor commit (which is the basis for the next set of changes).  This changes how we initially bootstrap the manager, and removes ERB template stuff in favor of a static ps1 file.

File: lib/puppet_x/templates/init_ps.ps1.erb (hunk 1 above)
Comment (Iristyle) [inline lib/puppet_x/templates/init_ps.ps1.erb]: I hid it behind `$DebugPreference` now.

File: lib/puppet_x/puppetlabs/powershell/powershell_manager.rb [hunk 2]
```diff
@@ -101,8 +101,9 @@ def self.init_path
       end
 
       def make_ps_init_code(init_ready_event_name)
+        debug_output = Puppet::Util::Log.level == :debug ? '$True' : '$False'
         <<-CODE
-. #{self.class.init_path} -InitReadyEventName '#{init_ready_event_name}'
+. #{self.class.init_path} -InitReadyEventName '#{init_ready_event_name}' -EmitDebugOutput #{debug_output}
```
Comment (Iristyle) [inline lib/puppet_x/puppetlabs/powershell/powershell_manager.rb]: @glennsarti I've given the user control over the output ... either via using `--debug` with Puppet or through `$DebugPreference`

**Comment by Iristyle** [comment]
@puppetlabs/windows can I get a merge?

**Comment by ferventcoder** [comment]
:+1:

**Pull request merged**